File: JAVA/Maximum sum of hour glass.java
class Solution {
    int findMaxSum(int n, int m, int mat[][]) {
        int R = mat.length, C = mat[0].length, res = -1;
        for (int i = 0, cur = 0; i <= R - 3; i++, cur = 0) {
            for (int j = 0; j <= C - 3; j++) {
                cur = mat[i][j] + mat[i][j + 1] + mat[i][j + 2] + mat[i + 1][j + 1] + mat[i + 2][j] + mat[i + 2][j + 1] + mat[i + 2][j + 2];
                res = Math.max(res, cur);
            }
        }
        return res;
    }
};

File: C++/Maximum sum of hour glass.cpp
class Solution {
  public:
    int findMaxSum(int n, int m, vector<vector<int>> mat) {
        if(n <= 2 || m <= 2) return -1;
        int res = 0;
        for(int i=1; i<n-1; i++){
            for(int j=1; j<m-1; j++){
                int sum = mat[i-1][j-1] + mat[i-1][j] + mat[i-1][j+1] 
                          + mat[i][j]                                 
                          + mat[i+1][j-1] + mat[i+1][j] + mat[i+1][j+1];
                res = max(res,sum);
            }
        }
        return res; 
    }
};
